table.go

```diff
@@ -112,6 +112,49 @@ func (rt *RoutingTable) NPeersForCpl(cpl uint) int {
 	}
 }
 
+// UsefulPeer verifies whether the given peer.ID would be a good fit for the routing table
+// It returns true if the bucket corresponding to peer.ID isn't full, if it contains
+// replaceable peers or if it is the last bucket and adding a peer would unfold it.
+func (rt *RoutingTable) UsefulPeer(p peer.ID) bool {
+	rt.tabLock.RLock()
+	defer rt.tabLock.RUnlock()
+
+	// bucket corresponding to p
+	bucketID := rt.bucketIdForPeer(p)
+	bucket := rt.buckets[bucketID]
+
+	// bucket isn't full
+	if bucket.len() < rt.bucketsize {
+		return true
+	}
+
+	// find and returns a replaceable peer (if any)
+	replaceablePeer := bucket.min(func(p1 *PeerInfo, p2 *PeerInfo) bool {
```

Comment:
I used the same snippet that is already used in https://github.com/libp2p/go-libp2p-kbucket/blob/06cb92315c23c9b81f61121137ab36766007ebbb/table.go#L214-L220

technically, the `.min` doesn't matter because we just need to know whether there exists at least 1 replaceable peer in the bucket.

I am happy to copy the comment from the other snippet if it adds clarity, or to iterate on the peers within the bucket if it makes more sense.

Reply:
Updated 
https://github.com/libp2p/go-libp2p-kbucket/blob/d29769f9a18f18fcd4eecd637078e4f0b1657dd8/table.go#L131-L138